presto-hive/src/main/java/com/facebook/presto/hive/RetryDriver.java

```diff
@@ -43,6 +44,7 @@
     private final Function<Exception, Exception> exceptionMapper;
     private final List<Class<? extends Exception>> exceptionWhiteList;
     private final Optional<Runnable> retryRunnable;
+    private final List<Throwable> suppressedExceptions = new ArrayList<>();
```

Comment:
This should be a variable inside `run`. So far there was nothing in RetryDriver preventing it from being used more than once.